src/Z3/Lock.hs

```diff
@@ -0,0 +1,75 @@
+-- |
+-- Module    : Z3.Lock
+-- Copyright : (c) Iago Abal, 2013-2015
```

Comment:
If the code is mostly a copy paste from concurrent-extra, then the copyright notice should be:
```haskell
-- Copyright  : (c) 2010-2011 Bas van Dijk & Roel van Dijk
```